src/Modal.js

```diff
@@ -122,7 +122,8 @@ class Modal extends React.Component {
       this._element = null;
     }
 
-    const classes = document.body.className.replace('modal-open', '');
+    // Use regex to prevent matching `modal-open` as part of a different class, e.g. `my-modal-opened`
+    const classes = document.body.className.replace(/(^| )modal-open( |$)/, '');
```

Comment:
Best to use word boundaries `\b` because this will turn `my-modal-opened modal-open modal-opened` into `my-modal-openedmodal-opened` which is not wanted. This is because it removes the space from both sides.
`/\bmodal-open\b/` would result in `my-modal-opened  modal-opened`. There is double space, but that isn't going to break anything (technically, the current implementation would result in double spaces as well).